tests/unit/test_data.py

```diff
@@ -73,13 +73,13 @@ def test_dataset_raises_on_initialisation_for_invalid_ranks(
 
 
 def test_dataset_getters() -> None:
-    query_points, observations = tf.zeros((3, 3)), tf.zeros((3, 3))
+    query_points, observations = tf.zeros((3, 2)), tf.ones((3, 1))
```

Comment:
this wasn't a very good test, without being able to differentiate between query points and observations, or axes